tests/test_temporal.py

```
@@ -3374,6 +3455,41 @@ def test_temporal_run_context_serializes_usage_limits():
+    ctx = RunContext(deps=None, model=TestModel(), usage=RunUsage(), run_id='run-123')
+    serialized = TemporalRunContext.serialize_run_context(ctx)
+    reconstructed = TemporalRunContext.deserialize_run_context(serialized, deps=None)
+
+    with pytest.raises(
+        UserError,
+        match=re.escape("'TemporalRunContext' object has no attribute 'model'. To make the attribute available"),
+    ):
+        _ = reconstructed.model
+
+    with pytest.raises(AttributeError, match="'TemporalRunContext' object has no attribute 'unknown_attribute'"):
+        getattr(reconstructed, 'unknown_attribute')
```

Ruff (0.15.21)

[warning] Do not call `getattr` with a constant attribute value. It is not any safer than normal property access.

Replace `getattr` with attribute access

(B009)